trace/collector.go

```diff
@@ -98,8 +98,9 @@ func (sc *UserTrafficCollector) Close() error {
 
 // This is a shim to add packet counts based on payload type.
 type PacketCountCollector struct {
-	PacketCounts PacketCountConsumer
-	Collector    Collector
+	PacketCounts         PacketCountConsumer
+	Collector            Collector
+	SendSuccessTelemetry func()
```

Comment:
Not really a fan of sending the telemetry function to this packet count collector. Any better ideas @mgritter ?

Reply:
This is an instance of the Collector interface, correct?  I would be worried about introducing a race condition because the same function is being called by the TelemetryWorker.

The alternative is having a channel that the TelemetryWorker listens to.  It would be nice to stop doing that after we got the first event.

The idiom "2. One receiver, N senders, the only receiver says "please stop sending more" by closing an additional signal channel" https://go101.org/article/channel-closing.html seems overkill for this, but perhaps we could use a sync.Once.

TelemetryWorker: listens to the channel
PacketCountCollector: sync.Once( write to channel, once both request and response count are nonzero )

Then all subsequent writes to the channel are skipped.

Reply:
Note that the sync.Once would have to be set up in apidump and shared among all these collectors.